subsys/bluetooth/controller/ll_sw/ull_adv_sync.c

```diff
@@ -161,8 +161,115 @@ uint8_t ll_adv_sync_param_set(uint8_t handle, uint16_t interval, uint16_t flags)
 uint8_t ll_adv_sync_ad_data_set(uint8_t handle, uint8_t op, uint8_t len,
 				uint8_t const *const data)
 {
-	/* TODO */
-	return BT_HCI_ERR_CMD_DISALLOWED;
+	struct pdu_adv_com_ext_adv *ter_com_hdr, *ter_com_hdr_prev;
+	struct pdu_adv_hdr *ter_hdr, ter_hdr_prev;
+	struct pdu_adv *ter_pdu, *ter_pdu_prev;
+	uint8_t ter_len, ter_len_prev;
+	uint8_t *ter_dptr, *ter_dptr_prev;
+	struct lll_adv_sync *lll_sync;
+	struct ll_adv_set *adv;
+	uint8_t ter_idx;
+
+	adv =  ull_adv_is_created_get(handle);
+	if (!adv) {
+		return BT_HCI_ERR_UNKNOWN_ADV_IDENTIFIER;
+	}
+
+	lll_sync = adv->lll.sync;
+	if (!lll_sync) {
+		return BT_HCI_ERR_UNKNOWN_ADV_IDENTIFIER;
+	}
+
+	/* Get reference to previous tertiary PDU data */
+	ter_pdu_prev = lll_adv_sync_data_peek(lll_sync);
+	ter_com_hdr_prev = (void *)&ter_pdu_prev->adv_ext_ind;
+	ter_hdr = (void *)ter_com_hdr_prev->ext_hdr_adi_adv_data;
+	ter_hdr_prev = *ter_hdr;
```

Comment:
I'm not sure I understand this line: You dereference a pointer and assign that another pointer? 

Shouldn't it have been 
```
ter_hdr_prev = ter_hdr;
```
?

Reply:
dereference a struct pointer  and assign to a struct (by-value). ter_hdr_prev  is used as read-only and later ter_hdr points to the double buffer that will get modified bit fields.

Reply:
But `ter_hdr_prev` is a pointer, not a struct. Should it have been `*ter_hdr_prev = *ter_hdr;` then?

Reply:
no it is not:
`struct pdu_adv_hdr *ter_hdr, ter_hdr_prev;`

Reply:
Ah, yes, you are correct. Had misread the type, as they were both defined on the same line. I would perhaps suggest declaring two variables of different types on different lines to avoid confusion :) You can resolve this comment